### PR TITLE
Feat/set duration

### PR DESCRIPTION
**Pull request by Bunnavoit**

### CODE REVIEW

File: contracts/EthLocker.sol [hunk 1]
```diff
@@ -1,17 +1,15 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.19;
 
-// Uncomment this line to use console.log
-// import "hardhat/console.sol";
-
 contract EthLocker {
     address public owner;
-    uint256 public constant lockDuration = 40;
+    uint256 public lockDuration;
```
Comment (hangleang) [inline contracts/EthLocker.sol:6]:
```suggestion
    uint256 public immutable lockDuration;
```

Comment (hangleang) [reply contracts/EthLocker.sol:6]: this can be set as immutable as we don't change it later

File: contracts/EthLocker.sol [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
     mapping(address => uint256) public deposits;
     mapping(address => uint256) public depositTimes;
 
     constructor() {
         owner = msg.sender;
+        lockDuration = 40;
```
Comment (hangleang) [inline contracts/EthLocker.sol:12]: accept user input

File: contracts/EthLocker.sol [hunk 3]
```diff
@@ -20,11 +18,15 @@ contract EthLocker {
     }
 
     modifier lockTimePassed() {
-        require(block.timestamp >= depositTimes[msg.sender] + lockDuration, "Lock time has not passed");
+        require(block.timestamp >= depositTimes[msg.sender] + lockDuration, "time has not passed yet");
         _;
     }
 
-    function deposit() external payable {
+    function setLockDuration(uint256 _lockDuration) external onlyOwner {
```
Comment (hangleang) [inline contracts/EthLocker.sol:25]: why do you do that? imagine if someone deposit at time X and lock duration Y, then at time X+1, owner suddenly increase the lock duration, what happen to the deposit?

File: contracts/EthLocker.sol [hunk 4]
```diff
@@ -50,8 +52,6 @@ contract EthLocker {
     }
 
     receive() external payable {
-        require(msg.value > 0, "Deposit amount must be greater than 0");
-        deposits[msg.sender] += msg.value;
-        depositTimes[msg.sender] = block.timestamp;
+        deposit();
```
Comment (hangleang) [inline contracts/EthLocker.sol:55]: correct ✅

File: scripts/deploy.js [hunk 5]
```diff
@@ -1,17 +1,16 @@
...
-main()
-  .then(() => process.exit(0))
-  .catch((error) => {
-    console.error(error);
-    process.exit(1);
-  });
+main().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
+});
```
Comment (hangleang) [inline scripts/deploy.js:16]: let use this new way of deploying smart contract: [Ignition module](https://hardhat.org/ignition/docs/getting-started#overview)

**Comment by hangleang** [comment]
closing for #1

File: contracts/EthLocker.sol (hunk 2 above)
Comment (Bunnavoit) [reply contracts/EthLocker.sol:12]: noted bong

**Pull request merged**